### PR TITLE
Some updates for HPL_GPU

### PR DESCRIPTION
**Pull request by lueelu**
We have completed the following updates:
1. Apply gpu-aware MPI to HPL_pdtrsv.cpp. Some hip/rocblas APIs were replaced with wrapper APIs.
2. Support P>1 in HPL.dat. To achieve this, we have added and initilized some extra params in panel structure to send the row-swapping information from host to device. 
3. Support inner ring bcast algorithms that broadcasts the panel to other columns.
4. Ensure the buffer in HPL_pdfact is allocated once to improve performance.
5. Add wrapper API of hipMemcpy2D.

### CODE REVIEW

File: src/comm/HPL_2rinM.cpp [hunk 1]
```diff
@@ -93,7 +93,7 @@ int HPL_binit_2rinM( PANEL )
 /*
  * Force the copy of the panel into a contiguous buffer
  */
-   HPL_copyL( PANEL );
+   // HPL_copyL( PANEL );
```
Comment (reger-men) [inline src/comm/HPL_2rinM.cpp]: This affects the CPU version!

File: src/comm/HPL_blonM.cpp [hunk 2]
```diff
@@ -84,7 +84,7 @@ int HPL_binit_blonM( PANEL )
 /*
  * Force the copy of the panel into a contiguous buffer
  */
-   HPL_copyL( PANEL );
+   // HPL_copyL( PANEL );
```
Comment (reger-men) [inline src/comm/HPL_blonM.cpp]: Same here "CPU version"!

File: src/comm/HPL_blong.cpp [hunk 3]
```diff
@@ -84,7 +84,7 @@ int HPL_binit_blong( PANEL )
 /*
  * Force the copy of the panel into a contiguous buffer
  */
-   HPL_copyL( PANEL );
+   // HPL_copyL( PANEL );
```
Comment (reger-men) [inline src/comm/HPL_blong.cpp]: Here!

File: src/pgesv/HPL_pdtrsv.cpp [hunk 4]
```diff
@@ -413,23 +335,17 @@ void HPL_pdtrsv
...
-#ifdef ROCM
-      hipMemcpy(dXR, XR, kbprev*sizeof(double), hipMemcpyHostToDevice);
-#endif
    }
 
 #ifdef ROCM
-   hipDeviceSynchronize();
+   HPL_BE_device_sync(T_HIP);
    if( Wfr  ) {
       hipHostFree( W  );
```
Comment (reger-men) [inline src/pgesv/HPL_pdtrsv.cpp]: Please use the BE to call all the HIP API. e.g. HPL_BE_host_free(...)

**Review by reger-men — COMMENTED** [review]
Please review the comments bellow.

**Comment by lueelu** [comment]
Corrected!

**Pull request merged**